src/creatures/monsters/monster.cpp

```diff
@@ -1050,6 +1055,34 @@ void Monster::onThinkSound(uint32_t interval) {
 	}
 }
 
+bool Monster::walkToSpawn() {
+	if (walkingToSpawn || !spawnMonster || !targetList.empty()) {
+		return false;
+	}
+
+	int32_t distance = std::max<int32_t>(Position::getDistanceX(position, masterPos), Position::getDistanceY(position, masterPos));
+	if (distance == 0) {
+		return false;
+	}
+
+	listWalkDir.clear();
+	if (!getPathTo(masterPos, listWalkDir, 0, std::max<int32_t>(0, distance - 5), true, true, distance)) {
+		return false;
+	}
+
+	walkingToSpawn = true;
+	startAutoWalk();
```

Comment:
Tente isto, compilei e foi ok, mas não testei in-game:
```suggestion
	stdext::arraylist<Direction> listDir(128);
	if (!getPathTo(masterPos, listDir, 0, std::max<int32_t>(0, distance - 5), true, true, distance)) {
		return false;
	}

	walkingToSpawn = true;
	startAutoWalk(listDir.data());
```

Reply:
Correto @dudantas 
Testado e funcional.